src/views/profile-view/ProfileAddLink.vue

```diff
@@ -0,0 +1,370 @@
+import { useUserStore } from "@/store/user";
+import { useValidation } from "@/utils/validation";
+import { Link, PerspectiveInput } from "@perspect3vism/ad4m";
+import { LinkType } from "datocms-structured-text-utils";
+import { ref } from "vue";
+import { defineComponent } from "vue-demi";
+import AvatarUpload from "@/components/avatar-upload/AvatarUpload.vue";
+import { NOTE_IPFS_EXPRESSION_OFFICIAL } from "@/constants/languages";
+
+type linkType = "community" | "channel" | "webLink" | null;
+export default defineComponent({
+  emits: ["cancel", "submit"],
+  setup() {
+    const linkType = ref<linkType>(null);
+    const step = ref(1);
+    const isAddLink = ref(false);
+    const newProfileImage = ref("");
+
+    const {
+      value: title,
+      error: titleError,
+      errorMessage: titleErrorMessage,
+      isValid: titleIsValid,
+      validate: validateTitle,
+    } = useValidation({
+      initialValue: "",
+      rules: [
+        {
+          check: (value: string) => (value ? false : true),
+          message: "Title is required",
+        },
+        {
+          check: (value: string) => value.length < 3,
+          message: "Should be 3 or more characters",
+        },
+      ],
+    });
+
+    const {
+      value: description,
+      error: descriptionError,
+      errorMessage: descriptionErrorMessage,
+      isValid: descriptionIsValid,
+      validate: validateDescription,
+    } = useValidation({
+      initialValue: "",
+      rules: [
+        {
+          check: (value: string) => (value ? false : true),
+          message: "Description is required",
+        },
+        {
+          check: (value: string) => value.length < 3,
+          message: "Should be 3 or more characters",
+        },
+      ],
+    });
+
+    // TODO: update validation rules for link
+    const {
+      value: link,
+      error: linkError,
+      errorMessage: linkErrorMessage,
+      isValid: linkIsValid,
+      validate: validateLink,
+    } = useValidation({
+      initialValue: "",
+      rules: [
+        {
+          check: (value: string) => (value ? false : true),
+          message: "link is required",
+        },
+        {
+          check: (value: string) => value.length < 3,
+          message: "Should be 3 or more characters",
+        },
+      ],
+    });
+
+    return {
+      linkType,
+      title,
+      titleError,
+      titleErrorMessage,
+      titleIsValid,
+      validateTitle,
+      description,
+      descriptionError,
+      descriptionErrorMessage,
+      descriptionIsValid,
+      validateDescription,
+      link,
+      linkError,
+      linkErrorMessage,
+      linkIsValid,
+      validateLink,
+      step,
+      isAddLink,
+      newProfileImage,
+    };
+  },
+  computed: {
+    canAddLink(): boolean {
+      return this.linkIsValid;
+    },
+    canCreateLink(): boolean {
+      return this.titleIsValid && this.descriptionIsValid;
+    },
+  },
+  methods: {
+    async createLink() {
```

Comment:
When creating an area lets also link to the flux://profile value by creating a link that looks like so:

```
{
    source: "flux://profile",
    predicate: "flux://has_area",
    target: "area-n"
}
```

This completes the profile graph as flux://profile as the root entry point. 
Also I know link predicate values are a complete mess now, sorry about that, we can go through and standardise everything soon.